### PR TITLE
fix: allow to call mockHttpclient multi times

### PR DESCRIPTION
**Pull request by fengmk2**
closes https://github.com/eggjs/egg-mock/issues/164

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/mock_httpclient.js [hunk 1]
```diff
@@ -93,8 +100,16 @@ module.exports = app => {
           return true;
         };
         pathname = path => {
-          return mockUrl.test(`${requestOrigin}${path}`);
+          for (const config of mockConfigs) {
+            if (config.mockUrl.test(`${requestOrigin}${path}`)) {
+              mm(app, '__httpclientNextMockConfigIndex', config.mockConfigIndex);
```
Comment (killagu) [inline lib/mock_httpclient.js:105]: 有并发请求的时候会不会有问题？

Comment (killagu) [reply lib/mock_httpclient.js:105]: 两次 mock，Promose.all 中的请求都匹配了。

Comment (fengmk2) [reply lib/mock_httpclient.js:105]: 只会命中第一个 mockUrl

File: test/mock_httpclient_next.test.js [hunk 2]
```diff
@@ -121,6 +121,48 @@ describe('test/mock_httpclient_next.test.js', () => {
...
 
+  it('should mockHttpclient call multi times work with Regex', async () => {
+    app.mockCsrf();
+    app.mockHttpclient(/\/not\/match\//, {
+      data: Buffer.from('mock not match response'),
+    });
+    app.mockHttpclient(/\/mock_url/, {
+      data: Buffer.from('mock 1 match response'),
+    });
+    app.mockHttpclient(/\/mock_url/, {
```
Comment (fengmk2) [inline test/mock_httpclient_next.test.js:132]: 加了相同 mock 校验，只会返回第一个 mock 结果

File: lib/mock_httpclient.js [hunk 3]
```diff
@@ -63,6 +63,9 @@ module.exports = app => {
    * @return {Context} this
    */
 
+  const MOCK_CONFIGS = Symbol('MOCK_CONFIGS');
+  const MOCK_CONFIG_INDEX = Symbol('MOCK_CONFIG_INDEX');
```
Comment (fengmk2) [inline lib/mock_httpclient.js:67]: 换成 Symbol 不被外部获取到

**Pull request merged**